compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -106,16 +106,18 @@ object ProtoTypes {
       if !res then ctx.typerState.constraint = savedConstraint
       res
 
-    /** Constrain result with special case if `meth` is a transparent inlineable method in an inlineable context.
-     *  In that case, we should always succeed and not constrain type parameters in the expected type,
-     *  because the actual return type can be a subtype of the currently known return type.
+    /** Constrain result with special case if `meth` is an inlineable method in an inlineable context.
+     *  In that case, if the method is transparent we should always succeed and not constrain type
+     *  parameters in the expected type, because the actual return type can be a subtype of the currently
+     *  known return type.
      *  However, we should constrain parameters of the declared return type. This distinction is
      *  achieved by replacing expected type parameters with wildcards.
      */
     def constrainResult(meth: Symbol, mt: Type, pt: Type)(using Context): Boolean =
-      if (Inlines.isInlineable(meth) && meth.is(Transparent)) {
-        constrainResult(mt, wildApprox(pt))
-        true
+      if (Inlines.isInlineable(meth)) {
+        val methodMatchedType = constrainResult(mt, wildApprox(pt))
+        // do not constrain the result type of transparent inline methods
+        meth.is(Transparent) || methodMatchedType
```

Comment:
I wrote this fix to get a green CI, however I am not convinced this is correct. Why do we have to use `wildApprox(pt)` in the case of non-transparent `inline` methods?

Reply:
I don't know. It seems non-sensical.

Reply:
So, probably the problem is somewhere else, and we should not change this part?